core/templates/domain/question/pretest-question-backend-api.service.spec.ts

```diff
@@ -133,4 +133,20 @@ describe('Pretest question backend API service', function() {
       expect(failHandler).toHaveBeenCalled();
     })
   );
+
+  it('should call successHandler with an empty array when storyId is empty',
+    fakeAsync(() => {
+      var successHandler = jasmine.createSpy('success');
+      var failHandler = jasmine.createSpy('fail');
+      var storyId = '';
```

Comment:
I don't think we expect storyID to be an empty string either it should be null of a non-empty storyID, @aks681 can you please confirm?

Also, do we have the test for the ID regex match?

@aks681, why these emptyID or regex match checks are inside the backendAPI service should we write this inside the directive/service before calling backend API service?

Reply:
Yeah I had the same question as @DubeySandeep. Why are we allowing storyId to be empty, and if its allowed to have storyId as empty, then why are we returning an empty array ?

Reply:
In the backend controller, we do throw an error if storyId is invalid. 
I think this is because, this function is called in `initExplorationPlayer`, so will be called when playing regular explorations as well. So, even if storyId is blank, it shouldn't throw an error in that case. This is also called inside a $q.all() in the service, so can't do conditions there unless everything is re-written. Since, a backend check is there against invalid storyIds, I think this check here is fine, though @wcascades, can you add an '_' to the regex check in the service, as I think underscores can also be present in a valid story id.

Reply:
@aks681 Ah, I was talking about the frontend backend-api.service. I mean why do we expect backend-api to do the checks? I think this check should be done before calling the backend API service i.e, in the directive/service which is calling this API.

@wcascades, Can you please add test for the regex check as well?

Reply:
Since this function is called in a `$q.all` in the directive along with the api calls to fetch exploration details. So, if a check is to be added there, the calling function would have to be rewriten. 
It could be moved there, by changing the array that $q.all() uses depending on whether storyId is present or not. @wcascades can you try that?